app/src/main/java/org/mifos/mobilebanking/presenters/SavingsMakeTransferPresenter.java

```diff
@@ -97,20 +98,20 @@ public void onNext(AccountOptionsTemplate accountOptionsTemplate) {
      * @param accountOptions {@link List} of {@link AccountOption}
      * @return Returns {@link List} containing {@code accountNo}
      */
-    public List<String> getAccountNumbers(List<AccountOption> accountOptions) {
-        final List<String> accountNumber = new ArrayList<>();
+    public List<AccountDetail> getAccountNumbers(List<AccountOption> accountOptions) {
+        final List<AccountDetail> accountNumber = new ArrayList<>();
         Observable.fromIterable(accountOptions)
-                .flatMap(new Function<AccountOption, Observable<String>>() {
+                .flatMap(new Function<AccountOption, Observable<AccountDetail>>() {
                     @Override
-                    public Observable<String> apply(AccountOption accountOption) {
-                        return Observable.just(accountOption.getAccountNo());
+                    public Observable<AccountDetail> apply(AccountOption accountOption) {
+                        return Observable.just(new AccountDetail(accountOption.getAccountNo(),
+                                accountOption.getAccountType().getValue()));
                     }
                 })
-                .distinct()
```

Comment:
@therajanmaurya , this was the cause of the crash while trying to open transfers from an loan account as it was filtering and removing the accounts with the same account number, but as you can see in the image we have account number as 0000001 for saving account and also for loan account. 
As we are searching directly from the accountOption template here:https://github.com/openMF/mifos-mobile/pull/760/files#diff-fd4b24c3db287adc9a965db1dec74947R307 but we are setting the index obtained from this to the spinner which is attached to the filtered list due to which it was throwing indexoutofbound exception.